feature_engine/prediction/target_mean_prediction.py

```diff
@@ -0,0 +1,271 @@
+        X_tr = self._pipeline.transform(X)
+
+        # calculate the average of each observation
+        predictions = X_tr[self.variables_].mean(axis=1)
+
+        return predictions
+
+    def _make_numerical_pipeline(self):
+        """
+        Create pipeline for a dataframe solely comprised of numerical variables
+        using a discretiser and encoder.
+        """
+        encoder = MeanEncoder(variables=self.variables_numerical_, errors="raise")
+
+        _pipeline_numerical = Pipeline(
+            [
+                ("discretisation", self._make_disretiser()),
+                ("encoder", encoder),
+            ]
+        )
+
+        return _pipeline_numerical
+
+    def _make_categorical_pipeline(self):
+        """
+        Instantiate the encoder for a dataframe solely comprised of categorical
+        variables.
+        """
+
+        return MeanEncoder(
+            variables=self.variables_categorical_, errors="raise"
+        )
+
+    def _make_combined_pipeline(self):
+
+        encoder_num = MeanEncoder(variables=self.variables_numerical_, errors="raise")
+        encoder_cat = MeanEncoder(variables=self.variables_categorical_, errors="raise")
+
+        _pipeline_combined = Pipeline(
+            [
+                ("discretisation", self._make_discretiser()),
+                ("encoder_num", encoder_num),
+                ("encoder_cat", encoder_cat),
+            ]
+        )
+
+        return _pipeline_combined
+
+    def _make_discretiser(self):
+        """
+        Instantiate either EqualWidthDiscretiser or EqualFrequencyDiscretiser.
+        """
+        if self.strategy == "equal-width":
+            discretiser = EqualWidthDiscretiser(
+                bins=self.bins, variables=self.variables_numerical_, return_object=True
+            )
+        else:
+            discretiser = EqualFrequencyDiscretiser(
+                q=self.bins, variables=self.variables_numerical_, return_object=True
+            )
+
+        return discretiser
+
+    def r_squared_score(self, X: pd.DataFrame, y: pd.Series,
+                        sample_weight: float = None) -> float:
+        """
+        Returns the coefficient of determination, which represents the proportion of
+        the variation in 'y' that can be predicted by 'X'.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            Test samples.
+
+        y : pandas series of shape = [n_samples, ]
+            True lables for 'X'.
+
+        sample_weight : pandas series of shape [n_samples, ], default=None
+            Sample weights.
+
+        Returns
+        --------
+        r2_score : float
+            Linear correlation between 'y' and 'y_pred'.
+
+        """
+        # Should we use the same or different nomenclature as sklearn?
+
+        y_pred = self.predict(X)
+        return r2_score(y, y_pred, sample_weight=sample_weight)
+
+    def mean_accuracy_score(self, X: pd.DataFrame, y: pd.Series,
```

Comment:
Let's unify methods `r_squared_score` and `mean_accuracy_score` into a method called `score()` so that it is the same name that in the sklearn ClassifierMixin or RegressorMixin.

We should have an additional init param, `regression=True`, that the user needs to set based on its target. If regression is true, then we use r2, if false then we use accuracy.

Also, we need to add the target checks based on the value of the init regression parameter like you did in the discretizer tree in a previous PR.